lldb/source/Plugins/LanguageRuntime/Swift/SwiftLanguageRuntimeImpl.h

```diff
@@ -256,6 +256,26 @@ class SwiftLanguageRuntimeImpl {
     stripSignedPointer(swift::remote::RemoteAbsolutePointer pointer) = 0;
   };
 
+  /// A wrapper around TargetReflectionContext, which holds a lock to ensure
+  /// exclusive access.
+  struct ThreadSafeReflectionContext {
+    ThreadSafeReflectionContext(ReflectionContextInterface *reflection_ctx,
+                             std::recursive_mutex &mutex)
+        : m_reflection_ctx(reflection_ctx), m_lock(mutex) {}
+
+    ReflectionContextInterface *operator->() const {
+      return m_reflection_ctx;
+    }
+
+    operator bool() const {
+      return m_reflection_ctx != nullptr;
+    }
+
+  private:
+    ReflectionContextInterface *m_reflection_ctx;
+    std::lock_guard<std::recursive_mutex> m_lock;
```

Comment:
Does this have to be a recursive mutex?

Reply:
I didn't see @adrian-prantl's comment, he answered it. Might be worth a comment still.

Reply:
+1 on the comment explaining this!